packages/main/src/SegmentedButton.ts

```diff
@@ -280,6 +333,14 @@ class SegmentedButton extends UI5Element {
 		if (parentWidth <= this.offsetWidth && this.absoluteWidthSet) {
```

Comment:
This can be optimized, also boolean flags that hold the opposite states should be (re)set properly based on the situation:

		if ((parentWidth <= parseInt(this.originalWidth) || parentWidth <= this.offsetWidth) && this.absoluteWidthSet) {
			this.style.width = "100%";
			this.percentageWidthSet = true;
			this.absoluteWidthSet = false; // <-- when set percentage to true, absolute should be false and vice versa
		} else if (parentWidth > parseInt(this.originalWidth)) {
			this.style.width = this.originalWidth;
			this.absoluteWidthSet = true;
			this.percentageWidthSet = false;
		}